### PR TITLE
Added producer reconnect logic to kafkahook

### PR DESCRIPTION
**Pull request by thrawn01**
## Purpose
Occasionally producers will encounter an error and kafka-hook will stop sending logs. When this happens `journalctl -u kafka-pixy@0.service` gets all the logs with the error `kafkahook buffer overflow`

## Implementation
* Reads from the errors channel to avoid producer blocking.

### CODE REVIEW

**Comment by horkhe** [comment]
It seems weird that producer does not reestablishes connection. There is no such problem in kafka-pixy by the way. It does not recreate producer. Maybe it was just an old version of Shopify/sarama that had a bug?

**Comment by thrawn01** [comment]
I was wondering about that, Yesterday while writing this code I looked at the KP code and couldn't find any re-connect logic. 

Could it simply be that we were not reading from the error channel and that was causing it to block when an error occurred? I noticed that `Return.Errors` is true by default. Perhaps disabling that will keep it from blocking. 

The more I think on it, I'm of the opinion we should log the error instead of disabling `Return.Errors`. I'll make changes to reflect that.

Thoughts?

**Comment by horkhe** [comment]
KP relies on Shopify/sarama reconnect logic. If `Return.Error` is `true` by default then we definitely should either set it to `false` or start reading from it. Not reading from the enabled errors channel causing producer to block.

File: kafkahook/kafkahook.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ func New(conf Config) (*KafkaHook, error) {
 	kafkaConfig.Producer.Flush.Frequency = 200 * time.Millisecond
 	kafkaConfig.Producer.Retry.Backoff = 10 * time.Second
 	kafkaConfig.Producer.Retry.Max = 6
+	kafkaConfig.Producer.Return.Errors = true
```
Comment (horkhe) [inline kafkahook/kafkahook.go]: It is true by default.

**Pull request merged**